third_party/xla_client/cache.h

```diff
@@ -90,6 +75,32 @@ class Cache {
   }
 
  private:
+  using ElementList = std::list<Element>;
+
+  struct Hasher {
+    size_t operator()(const K* key) const { return hasher(*key); }
+
+    H hasher;
+  };
+
+  struct Equaler {
+    bool operator()(const K* k1, const K* k2) const {
+      return equaler(*k1, *k2);
+    }
+
+    E equaler;
+  };
+
+  using ElementMap =
+      std::unordered_map<const K*, typename ElementList::iterator, Hasher,
+                         Equaler>;
+
+  void MaybeLRU(typename ElementList::iterator it) {
+    if (it != element_list_.begin()) {
```

Comment:
What is the harm in doing this unconditionally? I wouldn't expect a measurable performance hit from doing it unconditionally, plus you can then drop the `Maybe`. To be fair, you could drop the `Maybe` anyway, since it's already LRU on the no-op path.

Reply:
SG. Renamed DoLRU().